### PR TITLE
[AIRCoP] [CUDA] LogSoftmax and misc

### PR DESCRIPTION
**Pull request by supunab**
Adding support for logSoftmax (can be easily used to compute Cross Entropy Loss) and testing for both Softmax and LogSoftmax using PyTorch programs.

### CODE REVIEW

**Review by luke-jiang — APPROVED** [review]

File: src/main/scala/lms/thirdparty/cuda/cuda.scala [hunk 1]
```diff
@@ -1065,9 +1068,13 @@ trait CudaLibs extends CudaOps {
     cudaSyncThreads
   }
 
-  def softmax[N:Numeric:Manifest](log: Boolean)(implicit  __pos: SourceContext) = cudaGlobalDynamicFun {
+  def softmax[N:Numeric:Manifest](logSoftmax: Boolean)(implicit __pos: SourceContext) = cudaGlobalDynamicFun {
```
Comment (luke-jiang) [inline src/main/scala/lms/thirdparty/cuda/cuda.scala:1071]: maybe use `cudaSoftmax`, `cudaSoftmaxGrad` to keep naming consistency with other kernels?

Comment (supunab) [reply src/main/scala/lms/thirdparty/cuda/cuda.scala:1071]: yep, makes sense! will update. Thanks!
